packager/media/formats/mp4/low_latency_segment_segmenter.h

```diff
@@ -0,0 +1,69 @@
+// Copyright 2014 Google Inc. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file or at
+// https://developers.google.com/open-source/licenses/bsd
+
+#ifndef PACKAGER_MEDIA_FORMATS_MP4_LOW_LATENCY_SEGMENT_SEGMENTER_H_
+#define PACKAGER_MEDIA_FORMATS_MP4_LOW_LATENCY_SEGMENT_SEGMENTER_H_
+
+#include "packager/media/formats/mp4/segmenter.h"
+
+#include "packager/file/file.h"
+
+namespace shaka {
+namespace media {
+namespace mp4 {
+
+struct SegmentType;
+
+/// TODO(Caitlin): Write description
+/// Segmenter for LL-DASH profiles. There will be multiple
+/// media segments, which will contain multiple fragments. The generated segments
+/// are written as they are created to files defined by 
+/// @b MuxerOptions.segment_template if specified; otherwise, the segments are 
+/// appended to the main output file specified by @b MuxerOptions.output_file_name.
+class LowLatencySegmentSegmenter : public Segmenter {
+ public:
+  LowLatencySegmentSegmenter(const MuxerOptions& options,
+                        std::unique_ptr<FileType> ftyp,
+                        std::unique_ptr<Movie> moov);
+  ~LowLatencySegmentSegmenter() override;
+
+  /// @name Segmenter implementation overrides.
+  /// @{
+  bool GetInitRange(size_t* offset, size_t* size) override;
+  bool GetIndexRange(size_t* offset, size_t* size) override;
+  std::vector<Range> GetSegmentRanges() override;
+  /// @}
+
+ private:
+  // Segmenter implementation overrides.
+  Status DoInitialize() override;
+  Status DoFinalize() override;
+  Status DoFinalizeSegment() override;
+  Status DoFinalizeChunk() override;
+
+  // Write segment to file.
+  Status WriteInitSegment();
+  Status WriteChunk();
+  Status WriteInitialChunk();
+  Status FinalizeSegment();
+
+  uint64_t GetSegmentDuration();
+
+  std::unique_ptr<SegmentType> styp_;
+  uint32_t num_segments_;
+  bool is_initial_chunk_in_seg_ = true;
+  bool ll_dash_mpd_values_initialized_ = false;
+  File* segment_file_;
```

Comment:
It seems to me that this member should be a unique_ptr, to be cleaned up if the Segmenter is destroyed, rather than temporarily putting the handle into a local unique_ptr in the methods that use it, where it will only be cleaned up if there's an exception thrown or an early return while the local unique_ptr is in scope.

Reply:
I totally agree! :)